storage/tool/src/main/java/com/hortonworks/streamline/storage/tool/SQLScriptRunner.java

```diff
@@ -39,44 +42,99 @@
+        Class.forName(storageProperties.getDriverClass());
     }
 
-    public void runScript(String path, String delimiter) throws Exception {
-        final File file = new File(path);
-        if (!file.exists()) {
-            throw new RuntimeException("File not found for given path " + path);
+    public void runScriptWithReplaceDBType(String path) throws Exception {
+        String newPath = path.replace(PLACEHOLDER_REPLACE_DBTYPE, storageProperties.getDbType());
+        runScript(newPath);
+    }
+
+    public boolean checkConnection() {
+        try (Connection connection = connect()) {
+            return true;
+        } catch (SQLException e) {
+            System.out.println("Connection failure: " + e.getMessage());
+            return false;
         }
+    }
 
-        String content = new String(Files.readAllBytes(Paths.get(file.getAbsolutePath())), StandardCharsets.UTF_8);
+    private Connection connect() throws SQLException {
+        String user = storageProperties.getUser();
+        String password = storageProperties.getPassword();
+        String url = storageProperties.getUrl();
 
         Connection connection;
         if ((user == null && password == null) || (user.equals("") && password.equals(""))) {
-           connection =  DriverManager.getConnection(url);
+            connection = DriverManager.getConnection(url);
         } else {
             connection = DriverManager.getConnection(url, user, password);
         }
-        connection.setAutoCommit(true);
-
-        String[] queries = content.split(delimiter);
-        for (String query : queries) {
-            query = query.trim();
-            if (!query.isEmpty()) {
-                System.out.println(String.format("######## SQL Query:  %s ", query));
-                connection.createStatement().execute(query);
-                System.out.println("######## Query executed");
+        return connection;
+    }
+
+    public void runScript(String path) throws Exception {
+        String delimiter = storageProperties.getDelimiter();
+
+        final File file = new File(path);
+        if (!file.exists()) {
+            throw new RuntimeException("File not found for given path " + path);
+        }
+
+        String content = new String(Files.readAllBytes(Paths.get(file.getAbsolutePath())), StandardCharsets.UTF_8);
+
+        try (Connection connection = connect()) {
+            connection.setAutoCommit(true);
+
+            String[] queries = content.split(delimiter);
+            for (String query : queries) {
+                query = query.trim();
+                if (!query.isEmpty()) {
+                    System.out.println(String.format("######## SQL Query:  %s ", query));
+                    connection.createStatement().execute(query);
+                    System.out.println("######## Query executed");
+                }
             }
         }
     }
-    
+
+    public void downloadMySQLJarIfNeeded(String mysqlJarUrl) throws Exception {
```

Comment:
@satishd My understanding is postgres library is ASLv2 complaint where as mariadb or mysql driver is not. Hence we only added for mysql one. For postgres we should ship the library in classpath and shouldn't require a download.